projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLUtils.java

```diff
@@ -48,4 +51,31 @@ public static void validatePageInput(int pageSize, int pageNumber) {
       }
    }
 
+    /**
+     * This method counts Failed and Finished statuses for
+     * all data job executions for a given data job and
+     * returns an immutable pair of fail and success counts.
+     *
+     * @param dataJobName            the data job name.
+     * @param jobExecutionRepository the repository to query.
+     * @return ImmutablePair<Integer, Integer>, the left element is the Fail count, right Finished.
+     */
+    public static ImmutablePair<Integer, Integer> countFailedAndFinishedExecutions(String dataJobName,
```

Comment:
```suggestion
    public static Map<String, List<ExecutionStatus, Long>> countExecutionStatuses(List<String> dataJobNames,
```

Map<String, List<ExecutionStatus, Long>> = Map<JobName, List<Status, Count>>. In this way we will make this method more generic and can be reused for other calculations (e.g. count skipped executions for particular data jobs).